Sources/Hummingbird/Middleware/FileMiddleware.swift

```diff
@@ -90,11 +94,47 @@ where Provider.FileAttributes: FileMiddlewareFileAttributes {
         urlBasePath: String? = nil,
         cacheControl: CacheControl = .init([]),
         searchForIndexHtml: Bool = false
+    ) {
+        self.init(
+            fileProvider: fileProvider,
+            urlBasePath: urlBasePath,
+            cacheControl: cacheControl,
+            searchForIndexHtml: searchForIndexHtml,
+            mediaTypeFileExtensionMap: [:]
+        )
+    }
+
+    private init(
+        fileProvider: Provider,
+        urlBasePath: String? = nil,
+        cacheControl: CacheControl = .init([]),
+        searchForIndexHtml: Bool = false,
+        mediaTypeFileExtensionMap: [String: MediaType]
     ) {
         self.cacheControl = cacheControl
         self.searchForIndexHtml = searchForIndexHtml
         self.urlBasePath = urlBasePath.map { String($0.dropSuffix("/")) }
         self.fileProvider = fileProvider
+        self.mediaTypeFileExtensionMap = mediaTypeFileExtensionMap
+    }
+
+    public func withAdditionalMediaType(_ mediaType: MediaType, mappedToFileExtension fileExtension: String) -> FileMiddleware {
+        withAdditionalMediaTypes(forFileExtensions: [fileExtension: mediaType])
+    }
+
+    public func withAdditionalMediaTypes(forFileExtensions extensionToMediaTypeMap: [String: MediaType]) -> FileMiddleware {
```

Comment:
@adam-fowler Is it worth adding a custom `ExpressibleByDictionaryLiteral` type for this dictionary use case? That way we can represent it in, what I believe, would be a clearer API surface. That could also cascade into the `public var extensionMap: ... { get }` down the line

Reply:
We can look at this in another PR. We can merge this, but if we want a custom type we'll need that PR before we do a release.

Reply:
@mredig do you want to give that custom type a shot? I think it'd be a great addition.

Reply:
Yeah I can do another PR as @adam-fowler suggested, after this one. :)